org.graalvm.polyglot.Context & BinaryString

konikvranik opened the pull request:
- support for ES6 #101
- preparation for proper binary files handling #70

quangson91 commented:
@konikvranik Thank you so much for creating a PR.
Did you finish this PR?
I would like to check & merge.

Thanks!

konikvranik commented:
Hi @quangson91, you're free to review & merge. For next feature (#110 ) I've created extra PR. So I'll wait for you until I'll merge it into mine master (which is source for this PR)

quangson91 commented:
@konikvranik 
There is some unit test that failed.
Would you mind checking it?
Otherwise, I will check it later.

quangson91 commented on .github/workflows/build.yml:
```diff
@@ -30,7 +30,7 @@ jobs:
           java-version: 11
 
       - name: Clean & Build Jar
-        run: ./gradlew clean test jar
```
We should keep `test`.
So in the future, if there is change, we will easier to find the problem.

konikvranik replied:
BTW, in the gradle there is `check` task which should be wider than just the `test`. Build is including the  `check`

quangson91 replied:
Yup. You are right.

konikvranik commented:
I've just noticed. There is no reason to fail the test. Probably just single run failure. I removed the redundant workflow I added by mistake. Now it should pass.

quangson91 reviewed (commented):
Just have a small question.
Otherwise, LGTM!

konikvranik commented on src/main/java/uos/dev/restcli/HttpRequestFilesExecutor.kt:
```diff
@@ -122,6 +122,7 @@ class HttpRequestFilesExecutor constructor(
             val rawRequest = requests.getOrNull(requestIndex) ?: return
 
             runCatching {
+                jsClient = JsClient()
```
That's why with Graal everything was held on context. So if you created `const` in one of requests and then you tried torepeate it with the same name in the another, it start's to complain that this already exists. So now for every request must me extra context.

quangson91 commented:
@konikvranik I merged. Thank you so much!

konikvranik commented:
@quangson91 Thank you for the project. :-)
